synapse/federation/federation_base.py

```diff
@@ -90,11 +90,11 @@ async def _check_sigs_and_hash(
 
         if result:
             logger.warning(
-                "Event contains spam, redacting %s: %s",
+                "Event contains spam, soft-failing %s: %s",
                 pdu.event_id,
                 pdu.get_pdu_json(),
             )
-            return prune_event(pdu)
+            pdu.internal_metadata.soft_failed = True
```

Comment:
Why don't we prune the event anymore? Could be useful to do for db size management (say spammer is abusing the max event size), ymmv

Reply:
I guess so. it seemed a bit redundant.
eh, I'll stick the redaction back in.